### ISSUE TITLE
Bug: sort-objects - In function parameters, don't break references to earlier params

### ISSUE THREAD

**Issue by dylang** [azat-io/eslint-plugin-perfectionist#9]
### Describe the bug

I'm sorry, this is probably an unexpected side effect of adding `sort-objects` to function parameters! 

I think this is the only place this can occur. 

```ts
// Sorting these will break the code
const fn = ({ b, a = b}) => {};
function fn({ b, a = b}) {};
```

## Before

```ts
export const compileMonorepoDependencies = async ({
    packageJson: { dependencies },
    targetRootPath, 
    parentModulePath = targetRootPath // <-- gets the value from targetRootPath!
}: {
    packageJson: PackageJson;
    targetRootPath: string;
    parentModulePath?: string;
}) =>
```

## After

```ts
export const compileMonorepoDependencies = async ({
    packageJson: { dependencies },
    parentModulePath = targetRootPath, // <-- Parameter 'parentModulePath' cannot reference identifier 'targetRootPath' declared after it.
    targetRootPath
}: {
    packageJson: PackageJson;
    parentModulePath?: string;
    targetRootPath: string;
}) =>
```

### ESLint version

v

### ESLint Plugin Perfectionist version

v1.1.0

### Additional comments

We don't want to change the order of the two keys that depend on each other, that seems doable.

But the interface should be sorted in the same order as the object.

```ts
export const compileMonorepoDependencies = async ({
    packageJson: { dependencies },
    targetRootPath, 
    parentModulePath = targetRootPath // <-- Gets the value from targetRootPath!
}: {
    packageJson: PackageJson;
    parentModulePath?: string; // <-- Sorted alphabetically, but no longer in the same order as the object it represents.
    targetRootPath: string;
}) =>
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thanks for the issue. This is an important problem, I will fix it

**Comment by azat-io** [comment]
Fixed in 3e987ae7e6fa9fc21caa7e23fb8b17d9ece027f0

Will be released today

**Comment by azat-io** [comment]
Released in [v1.1.1](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v1.1.1)
